rethinkdb/tests/common.py

```diff
@@ -0,0 +1,207 @@
+CLIENT_USER = 'doggo'
+
+# TLS.
+
+TLS_SERVER = 'server1'  # type: ServerName
+TLS_DRIVER_KEY = os.path.join(HERE, 'data', 'tls', 'server.key')
+TLS_DRIVER_CERT = os.path.join(HERE, 'data', 'tls', 'server.pem')
+TLS_CLIENT_CERT = os.path.join(HERE, 'data', 'tls', 'client.pem')
+
+# Database content.
+
+DATABASE = 'doghouse'
+
+HEROES_TABLE = 'heroes'
+HEROES_TABLE_CONFIG = {
+    'shards': 1,
+    'replicas': {'primary': 1, 'eu': 1},
+    'primary_replica_tag': 'primary',
+}
+HEROES_TABLE_SERVERS = {'server1', 'server2'}  # type: Set[ServerName]
+HEROES_TABLE_PRIMARY_REPLICA = 'server1'  # type: ServerName
+HEROES_TABLE_REPLICAS_BY_SHARD = {0: HEROES_TABLE_SERVERS}
+HEROES_TABLE_DOCUMENTS = [
+    {
+        "hero": "Magneto",
+        "name": "Max Eisenhardt",
+        "aka": ["Magnus", "Erik Lehnsherr", "Lehnsherr"],
+        "magazine_titles": ["Alpha Flight", "Avengers", "Avengers West Coast"],
+        "appearances_count": 42,
+    },
+    {
+        "hero": "Professor Xavier",
+        "name": "Charles Francis Xavier",
+        "magazine_titles": ["Alpha Flight", "Avengers", "Bishop", "Defenders"],
+        "appearances_count": 72,
+    },
+    {
+        "hero": "Storm",
+        "name": "Ororo Monroe",
+        "magazine_titles": ["Amazing Spider-Man vs. Wolverine", "Excalibur", "Fantastic Four", "Iron Fist"],
+        "appearances_count": 72,
+    },
+]
+HEROES_TABLE_INDEX_FIELD = 'appearances_count'
+
+# Metrics lists.
+# NOTE: jobs metrics are not listed here as they're hard to trigger, so they're covered by unit tests instead.
+
+CONFIG_TOTALS_METRICS = (
+    (
+        'rethinkdb.server.total',
+        AggregatorStub.GAUGE,
+        lambda disconnected_servers: len(SERVERS) - len(disconnected_servers),
+        [],
+    ),
+    ('rethinkdb.database.total', AggregatorStub.GAUGE, 1, []),
+    ('rethinkdb.database.table.total', AggregatorStub.GAUGE, 1, ['database:{}'.format(DATABASE)]),
+    ('rethinkdb.table.secondary_index.total', AggregatorStub.GAUGE, 1, ['table:{}'.format(HEROES_TABLE)]),
+)  # type: Tuple[Tuple[str, int, Union[int, Callable[[set], int]], List[str]], ...]
+
+CLUSTER_STATISTICS_METRICS = (
+    ('rethinkdb.stats.cluster.queries_per_sec', AggregatorStub.GAUGE),
+    ('rethinkdb.stats.cluster.read_docs_per_sec', AggregatorStub.GAUGE),
+    ('rethinkdb.stats.cluster.written_docs_per_sec', AggregatorStub.GAUGE),
+)  # type: Tuple[Tuple[str, int], ...]
+
+SERVER_STATISTICS_METRICS = (
+    ('rethinkdb.stats.server.queries_per_sec', AggregatorStub.GAUGE),
+    ('rethinkdb.stats.server.queries_total', AggregatorStub.MONOTONIC_COUNT),
+    ('rethinkdb.stats.server.read_docs_per_sec', AggregatorStub.GAUGE),
+    ('rethinkdb.stats.server.read_docs_total', AggregatorStub.MONOTONIC_COUNT),
+    ('rethinkdb.stats.server.written_docs_per_sec', AggregatorStub.GAUGE),
+    ('rethinkdb.stats.server.written_docs_total', AggregatorStub.MONOTONIC_COUNT),
+    ('rethinkdb.stats.server.client_connections', AggregatorStub.GAUGE),
```

Comment:
If you add those structures in the code you could reuse them in the tests.

Reply:
I thought about it, but isn't it better that tests have separate data so that they serve as a double-check?